### DISCUSSION TITLE
Possible to add Tabulator as just another client_code Form?

### DISCUSSION THREAD

**Discussion by benlawraus** [anvilistas/tabulator#46] [Q&A]
Hi
I tried copying the client_code>Tabulator form into my anvil app so now it is seen as a custom component under client_code. I then import it as a custom component into another form.

Not surprisingly, that generated an error:

`AttributeError: module 'myApp.Tabulator' has no attribute 'modules'`

Can you let me know what more to do so that Tabulator can be imported in this way?

Why do I want to do this? I run pytests on my server and client code on my laptop and I need the Tabulator code to be present or else I will have to use a lot of mocks.

**Comment by s-cork** [comment]
I don't really follow the question.

Do you have a clone link that demonstrates the issue?

**Comment by benlawraus** [comment]
https://anvil.works/build#clone:N5XSFX4DQ6Q7SFGE=7IDQ7KL3DKG6ZKLJPPAM3CIY

Hi, after a bit more work, I got it right, I think. The above link shows a project with no dependency but  using Tabulator as a custom component form.
thanks for your hard work!
regards
Ben

**Reply by s-cork** [reply]
Ok great. Does that mean there's no issue now?

**Comment by benlawraus** [comment]
yes, all good
